ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -657,22 +679,111 @@ public void shouldInsertInto() throws Exception {
   }
 
   @Test
-  public void shouldHandleErrorResponseFromInsertInto() {
+  public void shouldPullQueryTableAfterInsert() throws Exception {
     // Given
     final KsqlObject insertRow = new KsqlObject()
-        .put("K", new KsqlObject().put("F1", new KsqlArray().add("my_key")))
+        .put("K", "my_key")
         .put("LONG", 11L);
 
     // When
+    client.insertInto(TEST_TABLE, insertRow).get();
+
+    // Then: cannot pull query table
+    final String query = "SELECT * from " + TEST_TABLE + " WHERE K='my_key';";
     final Exception e = assertThrows(
         ExecutionException.class, // thrown from .get() when the future completes exceptionally
-        () -> client.insertInto(AGG_TABLE, insertRow).get()
+        () -> client.executeQuery(query).get()
     );
 
     // Then
     assertThat(e.getCause(), instanceOf(KsqlClientException.class));
     assertThat(e.getCause().getMessage(), containsString("Received 400 response from server"));
-    assertThat(e.getCause().getMessage(), containsString("Cannot insert into a table"));
+    assertThat(e.getCause().getMessage(), containsString("table isn't queryable"));
+  }
+
+  @Test
+  public void shouldPushQueryTableAfterInsert() throws Exception {
+    // Given
+    final Map<String, Object> properties = new HashMap<>();
+    properties.put("auto.offset.reset", "earliest");
+
+    final KsqlObject insertRow = new KsqlObject()
+        .put("K", "my_key")
+        .put("LONG", 11L);
+
+    // When
+    final String query = "SELECT * from " + TEST_TABLE + " WHERE K='my_key' EMIT CHANGES LIMIT 1;";
+    StreamedQueryResult queryResult = client.streamQuery(query, properties).get();
+
+    final Row row = assertThatEventually(() -> {
+      // Potentially try inserting multiple times, in case the query wasn't started by the first time
+      try {
+        client.insertInto(TEST_TABLE, insertRow).get();
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+      return queryResult.poll(Duration.ofMillis(10));
+    }, is(notNullValue()));
+
+    // Then: a newly inserted row arrives
+    assertThat(row.getString("K"), is("my_key"));
+    assertThat(row.getLong("LONG"), is(11L));
+  }
+
+  @Test
+  public void shouldPullQueryCTASTableAfterInsert() throws Exception {
```

Comment:
Not sure if this test buys us much?

Reply:
Trying to make sure direct insert to CTAS table will have no effect

Reply:
What do you mean by "will not affected" exaclty?

Reply:
I mean you can insert to the table successfully but you won't see the result in pull request

Reply:
> I mean you can insert to the table successfully but you won't see the result in pull request

Well, while this is true, to me it's rather a design flaw / bug than a feature. Thus, I am not sure if it makes sense to test this case?

Reply:
+1 to Matthias: I don't think this test makes sense in this file. This test file is meant to test client-specific behavior. If the purpose of this test is to verify that INSERT VALUES does not update pull query state stores, then the test does not belong in this file. (In general, integration tests are (relatively) expensive so I'd err against adding redundant/duplicate tests.)

Reply:
Sure. I'll remove this